.github/workflows/ci.yml

```diff
@@ -18,24 +18,35 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Cache node modules
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: node_modules
-          key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
       - run: npm ci
       - run: npm run build
       - run: npm test
+      # Upload coverage for sonarcube
+      - uses: actions/upload-artifact@v3
+        with:
+          name: code-coverage
+          path: coverage/
 
+  sonar-scan:
+    needs: [unit-tests]
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    strategy:
+      matrix:
+        # Available OS's: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
+        os: [ubuntu-latest]
+        node-version: [16.x]
```

Comment:
Matrix can be deleted entirely now?

Reply:
Right, good point (and no need for node-version at all as well)